BibLaTeX.js

```diff
@@ -386,7 +386,12 @@ function creatorCheck(item, ctype) {
 	return false;
 }
 
-	function buildCiteKey(item, citekeys) {
+function buildCiteKey (item, extraFields, citekeys) {
+		const citationKey = extraFields.findIndex(field => field.field && field.value && field.field.toLowerCase() === 'citation key')
+		if (citationKey >= 0) return extraFields.splice(citationKey, 1)[0].value
+
+		if (item.citationKey) return item.citationKey
```

Comment:
shouldn't this be above the complex parsing for simplicity?

Reply:
You mean outside the buildCiteKey function? Or something else? Sorry, I don't quite follow, can you elaborate?

Reply:
Oh wait, do you mean first `item.citationKey` and then check the `extra` field? My idea was that if the user took the effort to put it in the `extra` field, that would always take precedence.

Reply:
yes, that's what I meant. I'm fine with leaving it as you have here. `item.citationKey` will never exists currently, though? Or has it already been generated here (sorry, didn't re-read code)? In that case, definitely go with the Extra solution as the first option as you have it.

Reply:
It's there in anticipation of citekeys arriving in Zotero as `.citationKey` (as discussed in https://github.com/zotero/translators/pull/1810#issuecomment-456219750) but in the interim, BBT will make its cite key available there.